packages/@aws-cdk/aws-rds/test/instance.test.ts

```diff
@@ -706,7 +706,45 @@ describe('instance', () => {
     instance.addRotationSingleUser();
 
     // THEN
-    expect(() => instance.addRotationSingleUser()).toThrow(/A single user rotation was already added to this instance/);
+    expect(() => instance.addRotationSingleUser()).toThrow(/A single user rotation for master secret is already added to this instance/);
+
+
+  });
+
+  test('throws when trying to add single user rotation for additional secret which is master secret', () => {
+    const instance = new rds.DatabaseInstance(stack, 'Database', {
+      engine: rds.DatabaseInstanceEngine.SQL_SERVER_EE,
+      instanceType: ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.SMALL),
+      credentials: rds.Credentials.fromUsername('syscdk'),
+      vpc,
+    });
+
+    // WHEN
+    instance.addRotationSingleUser();
+
+    // THEN
+    expect(() => instance.addRotationSingleUser({ secret: instance.secret!! })).toThrow(/A single user rotation for master secret is already added to this instance/);
+
+
+  });
+
+  test('throws when trying to add single user rotation for additional secret multiple times', () => {
+    const instance = new rds.DatabaseInstance(stack, 'Database', {
+      engine: rds.DatabaseInstanceEngine.SQL_SERVER_EE,
+      instanceType: ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.SMALL),
+      credentials: rds.Credentials.fromUsername('syscdk'),
+      vpc,
+    });
+    const secret = new rds.DatabaseSecret(stack, 'AdditionalSecret', {
+      username: 'myuser',
+    });
+    secret.attach(instance);
+
+    // WHEN
+    instance.addRotationSingleUser({ secret });
+
+    // THEN
+    expect(() => instance.addRotationSingleUser({ secret })).toThrow(/A single user rotation for additional secret is already added to this instance/);
```

Comment:
Same notes here as in `cluster.test.ts` (multiple lines, get rid of the blank lines, we're missing a positive-case test).

Reply:
Ditto.